LCM/dsc/engine/ConfigurationManager/RegistrationManagerHelper.c

```diff
@@ -108,15 +108,23 @@ MI_Result Register(
     MI_Uint32 getActionStatusCode;
     MI_Char* resultStatus = NULL;
     MI_Char* thumbprint = NULL;
+    MI_Value value;
+    MI_Uint32 flags;
     int systemResult = 0;
 
     if (cimErrorDetails)
     {
         *cimErrorDetails = NULL;
     }
-    
+
+
 #if defined(BUILD_OMS)
-    return MI_RESULT_OK;
+    // Check if RegistrationKey is specified. If not specified, do not attempt to register.
+    result = MI_Instance_GetElement(request->registrationData, MSFT_RegistrationKey_Name, &value, NULL, &flags, NULL);
+    if (result != MI_RESULT_OK || flags & MI_FLAG_NULL || value.string == NULL || value.string[0] == '\0')
+    {
+	return MI_RESULT_OK;
+    }
```

Comment:
Previously there was no chance of executing the code below.
Do you want to return a failure in an else clause here?